torch/nn/init.py

```diff
@@ -69,7 +73,7 @@ def _calculate_fan_in_and_fan_out(tensor):
     else:
         num_input_fmaps = tensor.size(1)
         num_output_fmaps = tensor.size(0)
-        receptive_field_size = np.prod(tensor.numpy().shape[2:])
+        receptive_field_size = reduce(mul, (tensor.numpy().shape[2:]))
```

Comment:
Can we remove the `.numpy()` call?
Also, we could probably remove the need of the `reduce` by selecting the first two dimensions and then using `.view(-1).size(0)` instead.